app/models/katello/candlepin/content.rb

```diff
@@ -65,8 +65,8 @@ def build_repository
           :label => label,
           :url => feed_url,
           :feed_ca => ca,
-          :feed_cert => product.certificate,
-          :feed_key => product.key,
+          :feed_cert => get_certificate_and_key(product, @content.modifiedProductIds)[:cert],
+          :feed_key => get_certificate_and_key(product, @content.modifiedProductIds)[:key],
```

Comment:
I'm not a fan of calling the method twice to get the same answer instead of assigning to a var. Up to you, though, if you'd prefer to keep.